### PR TITLE
feat(*): sync examples to v0.8 spec

### PR DESCRIPTION
**Pull request by lsytj0413**
Signed-off-by: lsytj0413 <511121939@qq.com>

**Many thanks for submitting your Pull Request :heart:!**

**What this PR does / why we need it**:

- sync the v0.8 specification examples

**Special notes for reviewers**:

**Additional information (if needed):**

- there is two example didn't synced:
  - https://github.com/serverlessworkflow/specification/tree/0.8.x/examples#book-lending
    - It doesn't have functions.json & events.json in examples
  - https://github.com/serverlessworkflow/specification/tree/0.8.x/examples#online-food-ordering

### CODE REVIEW

**Comment by lsytj0413** [comment]
1. we should keep the examples in a independent directory, and don't modify it, so we can synced the examples from specification repo easy
2. add a tool to convert json & yaml

File: model/workflow.go [hunk 1]
```diff
@@ -176,6 +178,17 @@ func (w *Workflow) UnmarshalJSON(data []byte) error {
 			if nestedData, err = getBytesFromFile(s); err != nil {
 				return err
 			}
+
+			// TODO: optimize this
+			// In the specification, we can declared independently definitions with another file format, so
+			// we must convert the independently yaml source to json before unmarshal.
```
Comment (ricardozanini) [inline model/workflow.go:184]: @lsytj0413 wdyt about opening an issue for this one?

File: parser/functiondefs.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        {
+            "name": "sendSuccessEmail",
+            "operation": "file://myapis/emailapis.json#paymentSuccess"
+        },
+        {
+            "name": "sendInsufficientFundsEmail",
+            "operation": "file://myapis/emailapis.json#paymentInsufficientFunds"
+        }
+    ]
+}
```
Comment (ricardozanini) [inline parser/functiondefs.json:16]: Can't we move these files to `testdata` directory?

File: parser/parser_test.go [hunk 3]
```diff
@@ -30,7 +31,12 @@ func TestBasicValidation(t *testing.T) {
 	assert.NoError(t, err)
 	for _, file := range files {
 		if !file.IsDir() {
+			if file.Name() == "paymentconfirmation.json" {
+				fmt.Printf("done")
+			}
```
Comment (ricardozanini) [inline parser/parser_test.go:36]: Why do we need this?

Comment (lsytj0413) [reply parser/parser_test.go:36]: aha，it's the test code and i forget to remove it ^^

File: model/workflow.go (hunk 1 above)
Comment (lsytj0413) [reply model/workflow.go:184]: nice advice, i will open an issue and fix it in another PR.

File: parser/functiondefs.json (hunk 2 above)
Comment (ricardozanini) [reply parser/functiondefs.json:16]: I think "2" could be a better option then because having these files lying in the package smells. At least in the `testdata` directory, we add purpose.

File: model/states.go [hunk 4]
```diff
@@ -139,8 +140,8 @@ type OperationState struct {
 
 // OperationStateTimeout ...
 type OperationStateTimeout struct {
-	StateExecTimeout  StateExecTimeout `json:"stateExecTimeout,omitempty"`
-	ActionExecTimeout string           `json:"actionExecTimeout,omitempty" validate:"omitempty,min=1"`
+	StateExecTimeout  *StateExecTimeout `json:"stateExecTimeout,omitempty"`
```
Comment (spolti) [inline model/states.go:143]: maybe you can revert this change as the timeouts will be fixed as part of https://github.com/serverlessworkflow/sdk-go/pull/104

**Comment by spolti** [comment]
small fixes for https://github.com/serverlessworkflow/sdk-go/issues/70

**Review by ricardozanini — APPROVED** [review]

**Comment by spolti** [comment]
@lsytj0413 can you please fix the conflicts?

**Comment by lsytj0413** [comment]
> @lsytj0413 can you please fix the conflicts?

https://github.com/serverlessworkflow/sdk-go/pull/105 should be merge first, i will resolve this conflict after 105 have merged.

cc @ricardozanini

**Comment by lsytj0413** [comment]
> It was merged, but we still have one file to solve the conflict.

Fixed，PTAL.

There are following actions:
- [ ] reorganize testdata to keep the examples in a independent directory (which synced from specification repo)
- [ ] change the root path of not absolute file to testdata directory

File: model/states.go [hunk 5]
```diff
@@ -292,8 +292,11 @@ func (j *EventBasedSwitchState) UnmarshalJSON(data []byte) error {
 		return err
 	}
 
-	if err := json.Unmarshal(eventBasedSwitch["timeouts"], &j.Timeouts); err != nil {
-		return err
+	eventBaseTimeoutsRawMessage, ok := eventBasedSwitch["timeouts"]
+	if ok {
```
Comment (spolti) [inline model/states.go:296]: I think it is not needed, right?

**Review by spolti — APPROVED** [review]
just a minor comment.

File: model/states.go (hunk 5 above)
Comment (ricardozanini) [reply model/states.go:296]: Why not?

**Review by ricardozanini — APPROVED** [review]

File: model/states.go (hunk 5 above)
Comment (lsytj0413) [reply model/states.go:296]: We must have this because the `timeouts` field is optional. If we don't have it, `checkcarvitals.json` and `eventbasedswitchstate.json` will failed with `unexpected end of JSON input` because we try to unmarshal empty string.
